extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallryeGraphqlProcessor.java

```diff
@@ -0,0 +1,434 @@
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.jandex.IndexView;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanDefiningAnnotationBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.LiveReloadBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.configuration.ConfigurationError;
+import io.quarkus.deployment.index.ClassPathArtifactResolver;
+import io.quarkus.deployment.index.ResolvedArtifact;
+import io.quarkus.deployment.util.FileUtil;
+import io.quarkus.deployment.util.ServiceUtil;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLRecorder;
+import io.quarkus.vertx.http.deployment.HttpRootPathBuildItem;
+import io.quarkus.vertx.http.deployment.RequireBodyHandlerBuildItem;
+import io.quarkus.vertx.http.deployment.RouteBuildItem;
+import io.quarkus.vertx.http.deployment.devmode.NotFoundPageDisplayableEndpointBuildItem;
+import io.quarkus.vertx.http.runtime.HandlerType;
+import io.smallrye.graphql.cdi.config.ConfigKey;
+import io.smallrye.graphql.cdi.config.GraphQLConfig;
+import io.smallrye.graphql.cdi.producer.GraphQLProducer;
+import io.smallrye.graphql.schema.Annotations;
+import io.smallrye.graphql.schema.SchemaBuilder;
+import io.smallrye.graphql.schema.model.Operation;
+import io.smallrye.graphql.schema.model.Reference;
+import io.smallrye.graphql.schema.model.Schema;
+import io.smallrye.graphql.spi.LookupService;
+import io.smallrye.graphql.spi.MetricsService;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Processor for Smallrye GraphQL.
+ * We scan all annotations and build the model during build.
+ * 
+ * @author Phillip Kruger (phillip.kruger@redhat.com)
+ */
+public class SmallryeGraphqlProcessor {
+
+    private static final Logger LOG = Logger.getLogger(SmallryeGraphqlProcessor.class.getName());
+    private static final String SCHEMA_PATH = "/schema.graphql";
+    private static final String SPI_PATH = "META-INF/services/";
+
+    @Inject
+    private LaunchModeBuildItem launchMode;
```

Comment:
Better push that as a parameter of your methods. It's better to keep track of dependencies.